### PR TITLE
fix(assignment-overview): duplicate assignment in overview,fix useEffect

### PR DESCRIPTION
**Pull request by bertyhell**
closes: https://district01.atlassian.net/browse/AVO2-877

* allow duplicate assignment from overview and also copy the collection if the assignment has one
* move duplication logic to service
* fix double fetch of assignments when loading the assignment-edit page

### CODE REVIEW

**Comment by benjaminnaesen** [comment]
`src/assignment/views/AssignmentEdit.tsx` moet echt wel dringend eens gerefactored worden. 1000 lines.

https://tenor.com/vdaO.gif

File: src/assignment/assignment.service.ts [hunk 1]
```diff
@@ -8,8 +8,15 @@ import { ApolloCacheManager } from '../shared/services/data-service';
 import toastService from '../shared/services/toast-service';
 import i18n from '../shared/translations/i18n';
 
+import { GET_COLLECTION_BY_ID } from '../collection/collection.gql';
+import { CollectionService } from '../collection/collection.service';
+import { navigate } from '../shared/helpers';
+import { ASSIGNMENT_PATH } from './assignment.const';
```
[the remark this answers is not in the document]
Comment (bertyhell) [reply src/assignment/assignment.service.ts]: yep, ik dacht dat tslint hierover waarschuwde
blijkbaar niet, ik heb de regels toegevoegd aan tsconfig: https://github.com/viaacode/avo2-client/pull/388/files#diff-e5e546dd2eb0351f813d63d1b39dbc48R19-R20

[the remark this answers is not in the document]
Comment (andrych) [reply src/assignment/assignment.service.ts]: de ingebouwde eslint van create-react-app vangt die ook op

File: src/assignment/views/AssignmentEdit.tsx [hunk 2]
```diff
@@ -334,7 +315,19 @@ const AssignmentEdit: FunctionComponent<AssignmentEditProps> = ({
 			setLoadingInfo,
 			t('assignment/views/assignment-edit___je-hebt-geen-rechten-om-deze-opdracht-te-bewerken')
 		);
-	}, [loadingInfo, location, match.params, setLoadingInfo, assignmentContent, t, user]);
+	}, [location, match.params, setLoadingInfo, setAssignmentContent, t, user]);
```
Comment (andrych) [inline src/assignment/views/AssignmentEdit.tsx]: `React Hook useEffect has missing dependencies: 'currentAssignment' and 'setBothAssignments'.`

**Pull request merged**